lib/mix_test_interactive/interactive_mode.ex

```diff
@@ -65,36 +65,38 @@ defmodule MixTestInteractive.InteractiveMode do
   end
 
   @impl GenServer
-  def handle_call({:command, command}, _from, state) do
+  def handle_call(:note_file_changed, _from, state) do
+    if state.settings.watching? do
+      run_tests(state)
+    end
+
+    {:reply, :ok, state}
+  end
+
+  @impl GenServer
+  def handle_cast({:command, command}, state) do
     case CommandProcessor.call(command, state.settings) do
       {:ok, new_settings} ->
-        {:reply, :ok, %{state | settings: new_settings}, {:continue, :run_tests}}
+        {:noreply, %{state | settings: new_settings}, {:continue, :run_tests}}
 
       {:no_run, new_settings} ->
         show_usage_prompt(new_settings)
-        {:reply, :ok, %{state | settings: new_settings}}
+        {:noreply, %{state | settings: new_settings}}
 
       :help ->
         show_help(state.settings)
-        {:reply, :ok, state}
+        {:noreply, state}
 
       :unknown ->
-        {:reply, :ok, state}
+        {:noreply, state}
 
       :quit ->
-        {:reply, :quit, state}
+        IO.puts("Shutting down...")
```

Comment:
There's a beautiful opportunity for a real show stopper function call: 🤩 

```suggestion
       show_stop()
```

Reply:
lol